content/spire/try/oidc-federation-aws.md

```diff
@@ -0,0 +1,415 @@
+
+| String | Description | Files to Change |
+| --- | --- | --- |
+| MY\_EMAIL\_ADDRESS | The terms of service for the Let's Encrypt certificate authority requires that you specify a valid email. Let's Encrypt is used to configure a certificate for the OIDC Discovery Domain (see the [Discovery Provider](#required-dns-a-record-for-the-oidc-discovery-provider-ip-address) info above). Example value: user@example.org | oidc-dp-configmap.yaml (1 instance) |
+| MY\_DISCOVERY\_DOMAIN | You must configure a public DNS A record for the IP address of the OIDC Discovery Provider. See the next section for details. Example value: `oidc-discovery.example.org` | ingress.yaml (2 instances), oidc-dp-configmap.yaml (1 instance), server-configmap.yaml (1 instance) |
+
+In the YAML files, instances of the `example.org` [trust domain](/spiffe/concepts/#trust-domain) are valid to use for this tutorial and do not need to be changed.
+
+## Deploy OIDC Discovery Provider Configmap
+
+The SPIRE OIDC Discovery Provider provides a URL to the location of the discovery document specified by the OIDC protocol. The oidc-dp-configmap.yaml file specifies the URL to the OIDC Discovery Provider.
+
+Before running the command below, ensure that you have replaced the MY\_DISCOVERY\_DOMAIN placeholder with the FQDN of the Discovery Provider as described in [Replace Placeholder Strings in YAML Files](#replace-placeholder-strings-in-yaml-files).
+
+Change to the directory containing the **k8s/oidc-aws** YAML files and use the following command to apply the updated server configmap, the configmap for the OIDC Discovery Provider, and deploy the updated **spire-server** statefulset:
+
+```console
+$ kubectl apply \
+    -f server-configmap.yaml \
+    -f oidc-dp-configmap.yaml \
+    -f server-statefulset.yaml
+```
+
+To verify that the spire-server-0 pod has spire-server and spire-oidc containers, run:
+
+```console
+$ kubectl get pods spire-server-0 -n spire -o jsonpath='{.spec.containers[*].name}{"\n"}'
+```
+
+## Configure the OIDC Discovery Provider Service and Ingress
+
+Use the following command to set up a service definition for the OIDC Discovery Provider and to configure ingress for that service:
+
+```console
+$ kubectl apply \
+    -f server-oidc-service.yaml \
+    -f ingress.yaml 
+```
+
+## Verify the spire-server Stateful Set
+
+Verify that the spire-server stateful set has been created:
+
+```console
+$ kubectl get statefulset --namespace spire
+
+NAME           READY   AGE
+spire-server   0/1     10h
+```
+
+## Verify That spire-oidc has an External Service Address
+
+Run the following command to verify that the EXTERNAL-IP value is present for the spire-oidc service. The spire-oidc Discovery Provider service must provide an external IP address for AWS to access the OIDC discovery document provided by spire-oidc.
+
+```console
+$ kubectl get service -n spire
```

Comment:
More specific versions of this command that get the user what they need:

`kubectl get service -n spire spire-oidc`
(shows just the spire-oidc service)

or

`kubectl get service -n spire spire-oidc -o jsonpath={'.status.loadBalancer.ingress[*].ip'}`
(shows just the external IP address of the spire-oidc service)

Reply:
Agreed. Changed doc to the first one.